WordPress/gutenberg
# Global Styles: Try style system at site edit

This is a fork and supersedes https://github.com/WordPress/gutenberg/pull/20061 by @itsjonq It adds a Global Styles sidebar with typography controls in the Site Editor.

![2020-03-30-global-styles-edit-site](https://user-images.githubusercontent.com/583546/77948908-6ca40100-72c6-11ea-8f73-1d7b4a6825f7.gif)

A few notes:

- The is not the final UI, we'll iterate in further PRs until we get to https://github.com/WordPress/gutenberg/issues/21030
- It focuses on the global typography first, leaving other things for subsequent PRs.
- It adds the mechanics to read/update user & theme data via the Entities API.
- It removes the use of the `wp-gs` class in favor of CSS variables in the blocks. Following suit on what was done for colors https://github.com/WordPress/gutenberg/pull/21021 and line-height https://github.com/WordPress/gutenberg/pull/20775 this PR adds the font-size & font-weight CSS variables to Paragraph and Heading blocks.
- It loads the CSS variables in both edit-site & edit-post ([see](https://github.com/WordPress/gutenberg/pull/20530#discussion_r400827213)).
- It leaves out the variable generation mechanism / variable calculation to further explore in a subsequent PR. See [comment](https://github.com/WordPress/gutenberg/pull/20530#discussion_r391767806).
- It uses the new interface package to add the global styles sidebar.

## Test instructions

Test in `edit-site`:

- Install and activate the [Global Styles demo theme](https://github.com/WordPress/theme-experiments/pull/22).
- Enable the FSE Experiment under "Gutenberg > Experiments".
- Go to "Gutenberg > Site Editor (beta)" and play with the Global Styles controls in the sidebar.
- Save the changes to the Global Styles template via the "Update design" button in the editor.
- Go to the front and see that your changes are applied and are the same as the editor.

Test in `edit-post`:

- Install and activate the [Global Styles demo theme](https://github.com/WordPress/theme-experiments/pull/22).
- Enable the FSE Experiment under "Gutenberg > Experiments".
- Go to the post editor and create some content (paragraphs & headings). Make sure the values saved in the previous test are honored (inspect in the browser devtools, perhaps).
- Edit the line-height of a paragraph or heading, and make sure it is updated.
